mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/search/NotDeletedWithRangeSearchOverride.java

```diff
@@ -75,9 +75,7 @@ public Flux<MessageUid> search(MailboxSession session, Mailbox mailbox, SearchQu
         SearchQuery.UidRange[] uidRanges = uidArgument.getOperator().getRange();
 
         return Flux.fromIterable(ImmutableList.copyOf(uidRanges))
```

Comment:
```suggestion
        return Flux.fromArray(uidRanges)
```
to avoid intermediate collection?